datadog/serializer_test.go

```diff
@@ -541,3 +553,69 @@ func BenchmarkAppendSanitizedMetricName(b *testing.B) {
 		}
 	})
 }
+
+// BenchmarkTrimComparison compares manual trimming vs bytes.TrimFunc
+func BenchmarkTrimComparison(b *testing.B) {
+	testCases := []struct {
+		name   string
+		prefix string
+		input  string
+	}{
+		{"no trim needed", "prefix.", "simple.metric"},
+		{"trim leading", "prefix.", "...trimmed"},
+		{"trim trailing", "prefix.", "trimmed..."},
+		{"trim both", "prefix.", "...trimmed..."},
+		{"trim all", "prefix.", "......"},
+		{"mixed trim chars", "prefix.", "._-test._-"},
+		{"typical metric", "myapp.", "http.server.duration"},
+	}
+
+	for _, tc := range testCases {
+		// Benchmark current manual approach
+		b.Run(tc.name+"/manual", func(b *testing.B) {
+			buf := make([]byte, 0, 512)
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
```

Comment:
```suggestion
			for b.Loop() {
```

`b.Loop` is the newer, improved way to do this since 1.24, and provides a `b.ResetTimer` for free ahead of the first iter. See https://pkg.go.dev/testing#B.Loop